### ISSUE TITLE
`object-curly-spacing`: space after `}` in `}?:` should not be needed with optional parameters

### ISSUE THREAD

**Issue by jeroen-hoekstra** [eslint-stylistic/eslint-stylistic#1214]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

So I updated @stylistic/eslint-plugin from `5.7.1` to `5.10.0` and now it finds some issues related to `object-curly-spacing`:

<img width="661" height="120" alt="Image" src="https://github.com/user-attachments/assets/0f5926fb-9dad-45b1-951e-41366bdc4f08" />

And when I put a space after the `}` another rule is complaining:

<img width="709" height="117" alt="Image" src="https://github.com/user-attachments/assets/dffef2b1-a6f5-4a2c-bbe4-f319d23d85cc" />

### Reproduction

Add the lines in a IDE

### System Info

```shell
"@stylistic/eslint-plugin": "5.10.0",
"@typescript-eslint/eslint-plugin": "8.59.4",
"@typescript-eslint/parser": "8.59.4",
"eslint": "9.39.2"

Linux 5.15 Ubuntu 22.04.4 LTS
Node: 24.6.0
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
